### PR TITLE
[Hub Generated] Review request for Microsoft.CostManagement to add version preview/2022-10-01-preview

### PR DESCRIPTION
**Pull request by thramu**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/thramu/azure-rest-api-specs/release-cost-management-Microsoft.CostManagement-2022-10-01-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by rkmanda** [comment]
>     "title": "CostManagementClient"

Please generate the PR by following the process outlined in the PR template

To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the wiki.

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:5 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/common-types.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-10-01",
+    "title": "Cost Management common types"
+  },
+  "paths": {},
+  "definitions": {
+    "ErrorResponse": {
```
Comment (TimLovellSmith) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/common-types.json:9]: >ErrorResponse [](http://example.com/codeflow?start=5&length=13)

Consider just leveraging the definitions from common-types instead:

[https://github.com/Azure/azure-rest-api-specs-pr/blob/main/specification/common-types/resource-management/v3/types.json#L336](https://github.com/Azure/azure-rest-api-specs-pr/blob/main/specification/common-types/resource-management/v3/types.json#L336)

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/common-types.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          "readOnly": true
+        },
+        "message": {
+          "description": "Error message indicating why the operation failed.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ProxyResource": {
```
Comment (TimLovellSmith) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/common-types.json:35]: >ProxyResource [](http://example.com/codeflow?start=5&length=13)

Consider inheritng the common-types v2 ProxyResource definition

[https://github.com/Azure/azure-rest-api-specs-pr/blob/main/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs-pr/blob/main/specification/common-types/resource-management/v2/types.json)

**Comment by TimLovellSmith** [comment]
>           "type": "string",

format uri?

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:2726 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             "Failed"

Can consider adding Canceled for future-proofing.

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:2705 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>       "description": "A Cost management REST API operation.",

capitalize - "cost management" ?

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:2448 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Operation id: {provider}/{resource}/{operation}.",

capitalize "Operation ID"

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:2452 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           ],

How come the description talks about both 'usage' and 'forecast' and 'UsageAndForecast' but forecast and UsageAndForecast are not in the enumeration values list??

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:1469 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "The time frame for pulling data for the report. If custom, then a specific time period must be provided.",

How must it be provided??

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:1476 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Chart type of the main view in Cost Analysis. Required.",

the main view in the cost analysis report?

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:1822 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         "taginheritance"

Should be PascalCase or CamelCase, not lowercase, right?

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:2952 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             "value": "taginheritance",

Should be PascalCase or CamelCase, not lowercase, right?

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/costmanagement.json:2959 in 5702906. [](commit_id = 57029061003c503034b71e5b2191c5d6ee8d62aa, deletion_comment = False)

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/examples/setting-delete.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "parameters": {
+    "api-version": "2022-10-01-preview",
+    "scope": "subscriptions/00000000-0000-0000-0000-000000000000",
+    "type": "taginheritance"
```
Comment (TimLovellSmith) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/examples/setting-delete.json:5]: >taginheritance [](http://example.com/codeflow?start=13&length=14)

Should be PascalCase or CamelCase, not lowercase, right?

File: custom-words.txt [hunk 4]
```diff
@@ -2724,3 +2724,4 @@ mediumrc
 smallrc
 Autotune
 autogrid
+taginheritance
```
Comment (TimLovellSmith) [inline custom-words.txt:2727]: Remove this. You won't need it if you use correct casing or the term 'tag inheritance' everywhere.

**Pull request closed without merging**